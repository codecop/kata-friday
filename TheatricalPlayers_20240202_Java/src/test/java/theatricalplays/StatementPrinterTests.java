package theatricalplays;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Map;

import static org.approvaltests.Approvals.verify;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {
        Map<String, Play> plays = Map.of(
                "hamlet",  new Play("Hamlet", Play.Type.TRAGEDY),
                "as-like", new Play("As You Like It", Play.Type.COMEDY),
                "othello", new Play("Othello", Play.Type.TRAGEDY));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.print(invoice, plays);

        verify(result);
    }

    @Test
    void statementWithNewPlayTypes() {
        Map<String, Play> plays = Map.of(
                "henry-v",  new Play("Henry V", Play.Type.HISTORY),
                "as-like", new Play("As You Like It", Play.Type.PASTORAL));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("henry-v", 53),
                new Performance("as-like", 55)));

        StatementPrinter statementPrinter = new StatementPrinter();
        Assertions.assertThrows(Error.class, () -> {
            statementPrinter.print(invoice, plays);
        });
    }
}
