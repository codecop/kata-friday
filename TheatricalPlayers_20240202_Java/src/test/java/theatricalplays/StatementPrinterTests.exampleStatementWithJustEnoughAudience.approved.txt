Statement for BigCo
  Hamlet: $410.00 (31 seats)
  As You Like It: $468.00 (21 seats)
  Othello: $410.00 (31 seats)
Amount owed is $1,288.00
You earned 6 credits
